experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnInit } from "@angular/core";
+import { FormControl, FormGroup, Validators } from "@angular/forms";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute, Router } from "@angular/router";
+import { ResponseStatus } from "trafficops-types";
+
+import { ServerService } from "src/app/api";
+import { DecisionDialogComponent, DecisionDialogData } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * StatusDetailsComponent is the controller for a status "details" page.
+ */
+@Component({
+	selector: "tp-status-details",
+	styleUrls: ["./status-details.component.scss"],
+	templateUrl: "./status-details.component.html",
+})
+export class StatusDetailsComponent implements OnInit {
+
+	/** Status ID expected from the route param using which we identify whether we are creating new status or load existing status */
+	public id: string | number | null = null;
```

Comment:
`null` doesn't represent a valid Status identifier, so I think it'd be best to not allow that to ever be the case.

Also if instead of using `string` you used a literal type i.e. `"new"`, it would make type checking much easier, because you would know that if it isn't the string `"new"` then it's not a string at all.

Reply:
done